client/src/components/StoreHeader/StoreHeader.tsx

```diff
@@ -1,31 +1,35 @@
-import React, { useEffect, useState } from "react";
-import { Store, emptyStore } from "src/interfaces";
-import { fetchJson } from "src/utils";
-import { STORE_API } from "src/constants";
+import React from "react";
+import { Store } from "src/interfaces";
+import { DEFAULT_BACK_BTN_TEXT } from "src/constants";
 
-function StoreHeader({ storeId }: { storeId: string | null }) {
-  const [curStore, setCurStore] = useState<Store>(emptyStore());
-
-  // fetch list of users
-  const fetchStore = async () => {
-    let data = {
-      "store-id": storeId,
-    };
-    const stores = await fetchJson("POST", data, STORE_API);
-    setCurStore(stores);
+function StoreHeader({
+  store,
+  link,
+  linkText,
+}: {
+  store: Store | null;
+  link?: string | null;
+  linkText?: string | null;
+}) {
+  const headerLink = () => {
+    if (link) {
```

Comment:
you can actually make use of react-router-dom and have it remember the history in the state. Then when you click back, it will take you to the previously visited page - all out of the box!

Reply:
Ah, I see, I'm currently looking into react-router-dom and refactoring how we link around our app as part of #57 :)